### PR TITLE
Put cwd of port server elsewhere

### PR DESCRIPTION
**Pull request by jtattermusch**
Should help with https://github.com/grpc/grpc/issues/4051.

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -759,9 +759,13 @@ def _start_port_server(port_server_port):
     env = dict(os.environ)
     env['BUILD_ID'] = 'pleaseDontKillMeJenkins'
     if platform.system() == 'Windows':
+      # Working directory of port server needs to be outside of Jenkins
+      # workspace to prevent file lock issues.
```
Comment (ctiller) [inline tools/run_tests/run_tests.py]: How about sys.environ['TEMP'] to avoid creating tooooo many directories?

Comment (jtattermusch) [inline tools/run_tests/run_tests.py]: The dir is only created once we start a new port server which should only happen once we bump its version.
As far as I remember, we had some issue with the TEMP env variable not being set in our win jenkins environment.

Comment (ctiller) [inline tools/run_tests/run_tests.py]: K... Let's bump the port_server version with this PR so the change gets picked up

Comment (jtattermusch) [inline tools/run_tests/run_tests.py]: Good point. Bumped the version.

Comment (ctiller) [inline tools/run_tests/run_tests.py]: Looks like it's failed: probably check the referenced log file for what went wrong

https://grpc-testing.appspot.com/job/gRPC_pull_requests/4117/config=dbg,language=c,platform=windows/console

**Comment by jtattermusch** [comment]
Odd things happending on Jenkins, I wonder if that's connected to some reference repo experiments.

File: tools/run_tests/run_tests.py (hunk 1 above)
Comment (jtattermusch) [inline tools/run_tests/run_tests.py]: Hm, the problem might be that the path is to the script is relative.

**Comment by jtattermusch** [comment]
I've fixed what needed to be fixed. PTAL.

The jenkins problems seem to be either failures to clone the repo (unrelated),  or race between executors on the same machine trying to kill and restart the port_server at the same time (and as expected just one of them succeeds).

**Pull request merged**